nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/autodiff/opvalidation/TransformOpValidation.java

```diff
@@ -1986,4 +1987,95 @@ public void testLogSumExp2(){
                     .gradientCheck(true));
         }
     }
+
+
+    @Test
+    public void testCRELU(){
+
+        Nd4j.getRandom().setSeed(12345);
+        INDArray inputArr = Nd4j.rand(DataType.DOUBLE, 2, 2);
+        SameDiff sd = SameDiff.create();
+        SDVariable in = sd.var(inputArr);
+
+        SDVariable crelu = new CReLU(sd,in).outputVariable();
+        INDArray expected = Nd4j.concat(1, Nd4j.nn.relu(inputArr, 0), Nd4j.nn.relu(inputArr.neg(), 0));
+
+
+        OpValidation.validate(new TestCase(sd)
+                .expectedOutput("crelu", expected)
+                .gradientCheck(true)
+        );
+    }
+
+    @Test
+    public void testClipByAvgNorm(){
+
+        Nd4j.getRandom().setSeed(12345);
+        INDArray inputArr = Nd4j.rand(DataType.DOUBLE, 2,2,2);
+        SameDiff sd = SameDiff.create();
+        SDVariable in = sd.var(inputArr);
+        SDVariable out = new ClipByAvgNorm(sd,in, 1e-2,0,1,2).outputVariable();
+        SDVariable expected = sd.math.clipByNorm(in, 1e-2,0,1,2).mul(inputArr.length());
+
+        out.markAsLoss();
+
+
+        OpValidation.validate(new TestCase(sd)
+                .expectedOutput("clipbyavgnorm", expected.eval())
+                .gradientCheck(false)
+        );
+    }
```

Comment:
For all of these tests, let's add a standard deviation op after the output. Something like:
```
SDVariable loss = sd.standardDeviation("loss", out, true);
loss.markAsLoss();
```

The reason: we are more likely to detect bugs using stdev than what's being done here.
When a loss function isn't specified, we by default use sum(last_output) where last_output = out = ClipByAvgNorm in this case.

The reason is simple:
if _out_ has shape [2,2] and loss = sum(out), then dL/dOut[i] = 1 * dSum/dOut[i] = 1
i.e., all the loss values dL/dOut[i] are the same, which won't pick up certain types of failures. Whereas with stdev, the gradients dL/dOut[i] will be all different.